worker/src/main/java/gov/cms/ab2d/worker/processor/JobProcessorImpl.java

```diff
@@ -355,13 +355,23 @@ private int processHandles(List<Future<Integer>> futureHandles, ProgressTracker
                     }
                     progressTracker.incrementProcessedCount();
                 } catch (InterruptedException e) {
+                    cancelFuturesInQueue(futureHandles);
                     final String errMsg = "interrupted exception while processing patient ";
                     log.error(errMsg);
                     throw new RuntimeException(errMsg, e);
+
                 } catch (ExecutionException e) {
-                    final String errMsg = "exception while processing patient ";
-                    log.error(errMsg, e);
+                    cancelFuturesInQueue(futureHandles);
+                    log.error("exception while processing patient ", e);
+
+                    final String errMsg;
+                    if (e.getCause() != null) {
+                        errMsg = e.getCause().getMessage();
+                    } else {
+                        errMsg = e.getMessage();
+                    }
```

Comment:
What if the cause has yet another cause? :)

Try https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/exception/ExceptionUtils.html#getRootCauseMessage-java.lang.Throwable-

Reply:
done - updated code to use the ExceptionUtils to get the root cause and message